### DISCUSSION TITLE
build-artifacts dict not being created

### DISCUSSION THREAD

**Discussion by aaron-hutton** [nasa/fprime#1450] [Q&A] [answered]
Does anyone know why the `build-artifacts/Linux/dict` directory wouldn't be created? The project happily compiles and I can run the executable, but the `dict` directory isn't being populated. `build-artifacts/Linux` only contains `bin` and `lib` directories. Without the `dict` directory `fprime-gds` fails to run.

I'm updating an old project to v3 using fpp models, as far as I know it should match up with `Ref` e.g. calling `register_fprime_deployment` in the main CMakeLists, although I must be missing something.

The project isn't located in the fprime directory, and was originally built using the [separated project example](https://github.com/fprime-community/separated-project-example), could that be causing problems?

**Comment by LeStarch** [comment, marked as answer]
I just ran into this one yesterday as well.  You must ensure several things are consistently names:

1. The folder Where your APP is defined e.g. `Ref/`
2. The project tag in the CMakeLists e.g. `project(Ref C CXX)`
3. The module name in: `Top/instances.fpp` and `Top/topology.fpp` e.g. `module Ref {`
4. Topology name in `Top/topology.fpp` e.g. `topology Ref {`

Likely one of these is miss-named: e.g. `RefDeployment` or `ref` (lowercase)

**Reply by LeStarch** [reply]
I am going to transfer this to an issue ticket because on our side we need to:

1. Do better checking to warn user of inconsistency
2. Update separated project example.
3. Document this behavior.

**Reply by aaron-hutton** [reply]
Yep, switching everything to be consistent fixed it. Thanks for the help
